harness/determined/util.py

```diff
@@ -144,6 +143,14 @@ def wrap_metrics(
         return {"metrics": metrics, "stop_requested": stop_requested, "invalid_hp": invalid_hp}
 
 
+def sizeof_dict(dict: Dict[str, Any]) -> int:
+    """
+    Serializes dict object to json and calculates size in bytes.
+    """
+    dict_json = json_encode(dict)
```

Comment:
were you able to see how long this operation took? Any noticeable performance impact to experiments?

Reply:
weren't any noticeable impacts when i ran the experiments. timed the harness in python, the entire sizeof_dict method takes around 7ms.